### ISSUE TITLE
Lack of color contrast between the completer type label and the background when the respective row is selected

### ISSUE THREAD

**Issue by joaopalmeiro** [jupyterlab/jupyterlab#16597]
Hi! 👋 

## Description

In the [code completion widget](https://jupyterlab.readthedocs.io/en/latest/user/completer.html), the completer type label lacks contrast with the background in the selected/candidate row:

<img width="1582" alt="Screenshot 2024-07-17 at 13 34 55" src="https://github.com/user-attachments/assets/13c50ede-d7cc-4fd5-a2d8-23515e96484e">

<img width="1582" alt="Screenshot 2024-07-17 at 13 34 37" src="https://github.com/user-attachments/assets/c7c18ae7-7083-43ea-973e-589337110ade">

<img width="1582" alt="Screenshot 2024-07-17 at 13 51 43" src="https://github.com/user-attachments/assets/4606f598-5dfb-4f45-ad6c-3fb866da55c6">

## Reproduce

1. Create a new notebook.
2. In the first cell, type `p` and press <kbd>Tab</kbd>.

## Expected behavior

The completer type label font color contrasts with the selected/candidate background color (at least [WCAG AA](https://webaim.org/resources/contrastchecker/)).

### Possible solution

Set the color to `white` or `var(--jp-ui-font-color0)` for the selected row label:

https://github.com/jupyterlab/jupyterlab/blob/152359f2d241d5760ea3bc8964ef9d2ab6b71124/packages/completer/style/base.css#L111-L131

```css
.jp-Completer-item.jp-mod-active .jp-Completer-typeExtended {
  color: white;
  /* or */
  color: var(--jp-ui-font-color0);
}
```

## Context

- Operating System and version: macOS Sonoma 14.5
- Browser and version: Chrome 126.0.6478.127
- JupyterLab version: 4.3.0a2
- Python version: 3.10.13

Let me know what you think and if I can open a PR. Thanks!

**Comment by krassowski** [comment]
Thank you!

>   color: var(--jp-ui-font-color0);

Sounds reasonable here if it works well in dark theme

**Comment by joaopalmeiro** [comment]
`--jp-ui-font-color0` for all themes (for the dark high contrast one is the same):

<img width="1582" alt="Screenshot 2024-07-17 at 21 32 29" src="https://github.com/user-attachments/assets/81f4bfef-5a5b-47b2-bc24-759c07636cc1">

<img width="1582" alt="Screenshot 2024-07-17 at 21 33 22" src="https://github.com/user-attachments/assets/5ab96c37-2dd8-47b0-986f-3b31307aa2cc">

- https://webaim.org/resources/contrastchecker/?fcolor=0000008A&bcolor=1976D2 vs. https://webaim.org/resources/contrastchecker/?fcolor=000000&bcolor=1976D2
- https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF8a&bcolor=2196f3 vs. https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=2196f3
- https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=319BF2

Better, but not perfect. The selection background color (blue) may also need to be adjusted. Either way, adjusting the foreground color would be an improvement. Thoughts, @krassowski?
